kythe/go/platform/kzip/kzip.go

```diff
@@ -198,6 +256,30 @@ func ReadConcurrency(n int) ScanOption {
 	return readConcurrency(n)
 }
 
+func (r *Reader) canonicalUnits() (string, []*zip.File) {
```

Comment:
If we allow only 1 representation in the file, why do we need separate directories?  Would a separate file extension for the entire kzip suffice (e.g. `.kzip.pb` or `.pb.kzip`)?  Or would a single metadata file at the root of the kzip suffice?

Reply:
We allow one or both representations to be present (at least based on this implementation).  It might be reasonable to indicate which is present in the extension at some point, though.

Reply:
Also, we need separate directories so that readers which only support the old format will exit early with an error when they encounter the new format (because they believe the kzip to be invalid), rather than only when they attempt to decode a CU.

Reply:
This is an important question. Is there value in having one file that is consumable by either a proto or JSON consumer? It's certainly simpler to have the file itself contain either one or the other; using the filename to indicate that seems reasonable to me (I'd vote for .pb.kzip or maybe pbkzip or kpbzip if the latter aren't too much of a mouthful). I do like not having to validate consistency across both representations.

If we do this, should our tooling (eg kzip) enforce the filename convention? (It doesn't now.) Maybe do a combination, something like file name is ".pb.kzip" and the directory is "units.pb", so if a file is fed to a system not expecting that encoding, it will just find no units (and presumably encounter an error)?

Reply:
Decided to keep a single file convention.